pkg/operator/configobserver/config_observer_controller.go

```diff
@@ -45,7 +45,7 @@ type Listers interface {
 // observedConfig that would cause the service being managed by the operator to crash. For example, if a required
 // configuration key cannot be observed, consider reusing the configuration key's previous value. Errors that occur
 // while attempting to generate the observedConfig should be returned in the errs slice.
-type ObserveConfigFunc func(listers Listers, existingConfig map[string]interface{}) (observedConfig map[string]interface{}, errs []error)
+type ObserveConfigFunc func(listers Listers, recorder events.Recorder, existingConfig map[string]interface{}) (observedConfig map[string]interface{}, errs []error)
```

Comment:
alternative: return slice of events.

Reply:
@sttts why you think that is better? ;)

Reply:
(it will mean every observe config func will have to use inmemory recorder and return events...), why not just create events directly?